torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1737,7 +1886,32 @@ std::vector<Val*> Index::getNonGlobalProducerStridedIndices(
 
   index_swizzle.run();
 
-  const auto& index_map = index_swizzle.indexMap();
+  auto producer_swizzled_index = index_swizzle;
+
+  if (producer_tv->hasSwizzleOp()) {
+    // Special handling needed on the new swizzle
+    //  op pass:
+    //  each swizzle op is local to the tensor,
+    //  so ReplayPasC will not include the swizzle
+    //  ops on the producer iterdomain. So would
+    //  need to traverse forward the producer domain
+    //  before the replay to get the swizzle ops.
+    IndexSwizzle producer_swizzle2d(
+        producer_tv,
+        domain_guard.prevDomain(),
+        producer_indexing.indexMap(),
+        producer_indexing.extentMap(),
+        producer_indexing.zeroDomains(),
+        producer_indexing.zeroMergedIn());
+    producer_swizzle2d.run();
+    producer_swizzled_index = producer_swizzle2d;
+  }
+
+  // TODO: merge the two swizzle compute logic once the new one is ready.
+  //  will need to replace cyclic shift swizzle with xor since swizzle2d
```

Comment:
Are there any advantages or disadvantages to doing so for transpose kernels? Seems it would be straightforward to add cyclic shift to swizzle2d.

Reply:
I guess not much difference. 

I can just add cyclic shift swizzle too when removing the legacy swizzle pass to keep the same behavior as before.

Reply:
I'm thinking of asking @zasdfgbnm to start looking at the point @rdspring1 got transpose scheduling. Seems we can try to see if it makes any difference. Xor swizzle instead of cyclic swizzle seems fine to me if there's no perf impact.